File: src/server/Adfnet.Service/Models/UserModel.cs
using System;
using System.Collections.Generic;
using Adfnet.Core;
using Adfnet.Core.ValueObjects;

namespace Adfnet.Service.Models
{
    public class UserModel : IServiceModel
    {
        public Guid Id { get; set; }
        public int DisplayOrder { get; set; }
        public bool IsApproved { get; set; }
        public int Version { get; set; }
        public DateTime CreationTime { get; set; }
        public IdName Creator { get; set; }
        public DateTime LastModificationTime { get; set; }
        public IdName LastModifier { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }
        public string Email { get; set; }
        public List<IdCodeNameSelected> Roles { get; set; }
        public string IdentityCode { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string DisplayName => FirstName + " " + LastName;
        public IdCodeName Language { get; set; }

    }
}

File: src/client/web/private/src/app/models/user-model.ts
import { IdCodeName } from '../value-objects/id-code-name';
import { IdCodeNameSelected } from '../value-objects/id-code-name-selected';
import { IdName } from '../value-objects/id-name';

export class UserModel {
    id: string;
    displayOrder: number;
    isApproved: boolean;
    version: number;
    creationTime: Date;
    creator: IdName;
    lastModificationTime: Date;
    lastModifier: IdName;
    username: string;
    password: string;
    email: string;
    identityCode: string;
    firstName: string;
    lastName: string;
    displayName: string;

    roles: IdCodeNameSelected[];
    language: IdName;
    languages: IdName[];
    constructor() {
        this.creator = new IdName();
        this.lastModifier = new IdName();
        this.language = new IdName();
        this.roles = [];
        this.languages = [];
    }
}
